src/h3lib/lib/algos.c

```diff
@@ -814,5 +814,6 @@ void H3_EXPORT(h3SetToLinkedGeo)(const H3Index* h3Set, const int numHexes,
     VertexGraph graph;
     h3SetToVertexGraph(h3Set, numHexes, &graph);
     _vertexGraphToLinkedGeo(&graph, out);
+    normalizeMultiPolygon(out);
```

Comment:
No check of the return value?

Reply:
There's no return value for `h3SetToLinkedGeo`, so I really wasn't sure what to do with the return value if it's > 0. There's no known case where this would be possible in any case - the rest of the algo should guarantee valid input.

Reply:
That is a problem... worst case scenario, you can set `errno`. If you find a scenario where this fails, either add a return value (which might be a breaking API change depending how you look at it), add a variant of this function that has a return value (`h3SetToLinkedGeoWithReturnCode`), or document that using such a polygon would result in undefined behavior.

Reply:
The issue here is that the user isn't passing in a polygon, they're passing in an array of hexagons. An array of hexagons with no dupes should *never* hit the error cases here, and if it does that's indicative of a problem in H3, not user error. I don't think it's reasonable to make all callers check for errors so that they can file bugs against our codebase if they get one.

I think it's possible for a user to hit the error cases if they pass in dupe hexagons, though I'm not sure. But this is possibly the worst way to check for that particular case.

Reply:
Ya. That's what I mean by "worst case scenario." I think your assumption that failure isn't possible makes sense for now. In the unlikely event that there is a situation that we haven't seen yet, then you still have a few options for emitting an error without breaking the API.

Reply:
I suggest in 4.0.0 we make essentially all functions able to indicate error. For the time being maybe add a comment that the error is being discarded?

Reply:
Will add. Were you thinking a user-facing doc comment, or just an inline comment?

Reply:
I think inline comment. I don't see a way a user could use this comment.

Reply:
There is a way to check that return values are not ignored in many compilers: ` __attribute__ ((warn_unused_result))`. If you do that on the declaration of `normalizeMultiPolygon`, you will get compiler warnings every time you compile this function. Not necessarily what you want, but something.